spectator/registry.py

```diff
@@ -1,286 +1,82 @@
-        self._client.post_json(self._uri, json)
-
-    def _publish(self):
-        snapshot = self._get_measurements()
-
-        if logger.isEnabledFor(logging.DEBUG):
-            for id, value in snapshot:
-                logger.debug("reporting: %s => %f", id, value)
-
-        if self._uri is not None:
-            i = 0
-            while i < len(snapshot):
-                end = min(i + self._batch_size, len(snapshot))
-                self._send_batch(snapshot[i:end])
-                i += self._batch_size
-
-    def _should_send(self, id, value):
-        max_op = 10
-        op = self._operation(id.tags())
-        return not math.isnan(value) and (value > 0 or op == max_op)
-
-    def _build_string_table(self, payload, data):
-        strings = {'name': 0}
-        for k, v in self._common_tags.items():
-            strings[k] = 0
-            strings[v] = 0
-
-        for id, _ in data:
-            strings[id.name] = 0
-            for k, v in id.tags().items():
-                strings[k] = 0
-                strings[v] = 0
-        keys = list(strings.keys())
-        keys.sort()
-        payload.append(len(keys))
-        payload.extend(keys)
-        for i, k in enumerate(keys):
-            strings[k] = i
-        return strings
+    def clock(self) -> Clock:
+        return self._clock
 
-    def _measurements_to_json(self, data):
-        payload = []
-        strings = self._build_string_table(payload, data)
-        for id, v in data:
-            self._append_measurement(strings, payload, id, v)
-        return payload
+    def age_gauge(self, name: str, tags: Optional[dict] = None) -> AgeGauge:
+        return AgeGauge(self._new_meter(name, tags), self._spectatord)
 
-    def _append_measurement(self, strings, payload, id, value):
-        tags = id.tags()
-        op = self._operation(tags)
-        common_tags = self._common_tags
-        payload.append(len(tags) + 1 + len(common_tags))
-        for k, v in common_tags.items():
-            payload.append(strings[k])
-            payload.append(strings[v])
-        for k, v in tags.items():
-            payload.append(strings[k])
-            payload.append(strings[v])
-        payload.append(strings["name"])
-        payload.append(strings[id.name])
-        payload.append(op)
-        payload.append(value)
+    def counter(self, name: str, tags: Optional[dict] = None) -> Counter:
+        return Counter(self._new_meter(name, tags), self._spectatord)
 
-    def _operation(self, tags):
-        if tags.get('statistic') in self.counterStats:
-            return self.addOp
-        else:
-            return self.maxOp
+    def distribution_summary(self, name: str, tags: Optional[dict] = None) -> DistributionSummary:
+        return DistributionSummary(self._new_meter(name, tags), self._spectatord)
 
+    def pct_distribution_summary(self, name: str,
+                                 tags: Optional[dict] = None) -> PercentileDistributionSummaryV2:
```

Comment:
Typically I would expect the return type to the the same as a normal distribution summary. Other than when creating them there shouldn't be any difference in usage and any code should be able to work with any distribution summary type without caring about the details.

Reply:
This is fixed - the `DistributionSummary` and `Timer` classes can each now handle both the standard and percentile variations of the meter types.